### PR TITLE
layers: Update Valid Usage enums (VL-61, partial)

### PR DESCRIPTION
**Pull request by mikew-lunarg**
Update Valid Usage enums in a subsection of core_validation.cpp by
adding a parameter to helper functions insideRenderPass() and
outsideRenderPass().  Update that VU status in the database.

Fix a couple other typos.

This partially implements Jira task VL-61.

Change-Id: I8d1fe99a237c4cd22216a1236f47f72fffd4fd00

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -4047,27 +4047,29 @@ static bool synchAndPrintDSConfig(layer_data *my_data, const VkCommandBuffer cb)
...
+static bool insideRenderPass(const layer_data *my_data, GLOBAL_CB_NODE *pCB, const char *apiName,
+                             UNIQUE_VALIDATION_ERROR_CODE msgCode) {
     bool inside = false;
     if (pCB->activeRenderPass) {
         inside = log_msg(my_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_BUFFER_EXT,
-                         (uint64_t)pCB->commandBuffer, __LINE__, DRAWSTATE_INVALID_RENDERPASS_CMD, "DS",
-                         "%s: It is invalid to issue this call inside an active render pass (0x%" PRIxLEAST64 ")", apiName,
-                         (uint64_t)pCB->activeRenderPass->renderPass);
+                         (uint64_t)pCB->commandBuffer, __LINE__, msgCode, "DS",
+                         "%s: It is invalid to issue this call inside an active render pass (0x%" PRIxLEAST64 ") %s", apiName,
```
Comment (tobine) [inline layers/core_validation.cpp]: Add a period after closing paren to separate custom message from generated message.  `..." PRIxLEAST64 "). %s",`

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -8808,7 +8811,7 @@ CmdFillBuffer(VkCommandBuffer commandBuffer, VkBuffer dstBuffer, VkDeviceSize ds
         cb_node->validate_functions.push_back(function);
 
         skip_call |= addCmd(dev_data, cb_node, CMD_FILLBUFFER, "vkCmdFillBuffer()");
-        skip_call |= insideRenderPass(dev_data, cb_node, "vkCmdCopyFillBuffer()");
+        skip_call |= insideRenderPass(dev_data, cb_node, "vkCmdFillBuffer()", VALIDATION_ERROR_01142);
```
Comment (tobine) [inline layers/core_validation.cpp]: woops good catch

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -7494,7 +7497,7 @@ VKAPI_ATTR VkResult VKAPI_CALL EndCommandBuffer(VkCommandBuffer commandBuffer) {
         if ((VK_COMMAND_BUFFER_LEVEL_PRIMARY == pCB->createInfo.level) || !(pCB->beginInfo.flags & VK_COMMAND_BUFFER_USAGE_RENDER_PASS_CONTINUE_BIT)) {
             // This needs spec clarification to update valid usage, see comments in PR:
             // https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/pull/516#discussion_r63013756
-            skip_call |= insideRenderPass(dev_data, pCB, "vkEndCommandBuffer");
+            skip_call |= insideRenderPass(dev_data, pCB, "vkEndCommandBuffer", VALIDATION_ERROR_00123);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens after function name for consistency `vkEndCommandBuffer()`

File: layers/core_validation.cpp [hunk 4]
```diff
@@ -8037,7 +8040,7 @@ VKAPI_ATTR void VKAPI_CALL CmdDraw(VkCommandBuffer commandBuffer, uint32_t verte
         if (!skip_call) {
             updateResourceTrackingOnDraw(pCB);
         }
-        skip_call |= outsideRenderPass(dev_data, pCB, "vkCmdDraw");
+        skip_call |= outsideRenderPass(dev_data, pCB, "vkCmdDraw", VALIDATION_ERROR_01365);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens

File: layers/core_validation.cpp [hunk 5]
```diff
@@ -8065,7 +8068,7 @@ VKAPI_ATTR void VKAPI_CALL CmdDrawIndexed(VkCommandBuffer commandBuffer, uint32_
         if (!skip_call) {
             updateResourceTrackingOnDraw(pCB);
         }
-        skip_call |= outsideRenderPass(dev_data, pCB, "vkCmdDrawIndexed");
+        skip_call |= outsideRenderPass(dev_data, pCB, "vkCmdDrawIndexed", VALIDATION_ERROR_01372);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens

File: layers/core_validation.cpp [hunk 6]
```diff
@@ -8148,7 +8151,7 @@ VKAPI_ATTR void VKAPI_CALL CmdDispatch(VkCommandBuffer commandBuffer, uint32_t x
         skip_call |= validate_and_update_draw_state(dev_data, pCB, false, VK_PIPELINE_BIND_POINT_COMPUTE, "vkCmdDispatch");
         skip_call |= markStoreImagesAndBuffersAsWritten(dev_data, pCB);
         skip_call |= addCmd(dev_data, pCB, CMD_DISPATCH, "vkCmdDispatch()");
-        skip_call |= insideRenderPass(dev_data, pCB, "vkCmdDispatch");
+        skip_call |= insideRenderPass(dev_data, pCB, "vkCmdDispatch", VALIDATION_ERROR_01562);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens

File: layers/core_validation.cpp [hunk 7]
```diff
@@ -9030,7 +9033,7 @@ CmdSetEvent(VkCommandBuffer commandBuffer, VkEvent event, VkPipelineStageFlags s
     GLOBAL_CB_NODE *pCB = getCBNode(dev_data, commandBuffer);
     if (pCB) {
         skip_call |= addCmd(dev_data, pCB, CMD_SETEVENT, "vkCmdSetEvent()");
-        skip_call |= insideRenderPass(dev_data, pCB, "vkCmdSetEvent");
+        skip_call |= insideRenderPass(dev_data, pCB, "vkCmdSetEvent", VALIDATION_ERROR_00238);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens

File: layers/core_validation.cpp [hunk 8]
```diff
@@ -9058,7 +9061,7 @@ CmdResetEvent(VkCommandBuffer commandBuffer, VkEvent event, VkPipelineStageFlags
     GLOBAL_CB_NODE *pCB = getCBNode(dev_data, commandBuffer);
     if (pCB) {
         skip_call |= addCmd(dev_data, pCB, CMD_RESETEVENT, "vkCmdResetEvent()");
-        skip_call |= insideRenderPass(dev_data, pCB, "vkCmdResetEvent");
+        skip_call |= insideRenderPass(dev_data, pCB, "vkCmdResetEvent", VALIDATION_ERROR_00249);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens

File: layers/core_validation.cpp [hunk 9]
```diff
@@ -9671,7 +9674,7 @@ CmdResetQueryPool(VkCommandBuffer commandBuffer, VkQueryPool queryPool, uint32_t
         } else {
             skip_call |= report_error_no_cb_begin(dev_data, commandBuffer, "vkCmdResetQueryPool()");
         }
-        skip_call |= insideRenderPass(dev_data, pCB, "vkCmdQueryPool");
+        skip_call |= insideRenderPass(dev_data, pCB, "vkCmdResetQueryPool", VALIDATION_ERROR_01025);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens

File: layers/core_validation.cpp [hunk 10]
```diff
@@ -10877,7 +10880,7 @@ CmdBeginRenderPass(VkCommandBuffer commandBuffer, const VkRenderPassBeginInfo *p
             }
             skip_call |= VerifyRenderAreaBounds(dev_data, pRenderPassBegin);
             skip_call |= VerifyFramebufferAndRenderPassLayouts(dev_data, cb_node, pRenderPassBegin);
-            skip_call |= insideRenderPass(dev_data, cb_node, "vkCmdBeginRenderPass");
+            skip_call |= insideRenderPass(dev_data, cb_node, "vkCmdBeginRenderPass", VALIDATION_ERROR_00440);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens

File: layers/core_validation.cpp [hunk 11]
```diff
@@ -10911,7 +10914,7 @@ VKAPI_ATTR void VKAPI_CALL CmdNextSubpass(VkCommandBuffer commandBuffer, VkSubpa
     if (pCB) {
         skip_call |= validatePrimaryCommandBuffer(dev_data, pCB, "vkCmdNextSubpass");
         skip_call |= addCmd(dev_data, pCB, CMD_NEXTSUBPASS, "vkCmdNextSubpass()");
-        skip_call |= outsideRenderPass(dev_data, pCB, "vkCmdNextSubpass");
+        skip_call |= outsideRenderPass(dev_data, pCB, "vkCmdNextSubpass", VALIDATION_ERROR_00458);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens

File: layers/core_validation.cpp [hunk 12]
```diff
@@ -10973,7 +10976,7 @@ VKAPI_ATTR void VKAPI_CALL CmdEndRenderPass(VkCommandBuffer commandBuffer) {
                 }
             }
         }
-        skip_call |= outsideRenderPass(dev_data, pCB, "vkCmdEndRenderpass");
+        skip_call |= outsideRenderPass(dev_data, pCB, "vkCmdEndRenderpass", VALIDATION_ERROR_00464);
```
Comment (tobine) [inline layers/core_validation.cpp]: Add open/close parens

**Comment by mikew-lunarg** [comment]
Thanks.  Made the requested tweaks.

**Pull request merged**